### ISSUE TITLE
[Gemma-3-27b-it] [WH T3K]: RULER eval crashes with DRAM OOM during batched prefill

### ISSUE THREAD

**Issue by arobergeTT** [tenstorrent/tt-inference-server#2389]
### Summary

RULER evals for `gemma-3-27b-it` on T3K crash with a DRAM Out of Memory error when the scheduler prefills two ~31K-token requests simultaneously. The vLLM engine dies and all subsequent requests fail with HTTP 500.

The server starts successfully and processes initial (shorter context) RULER requests. After ~27/175 requests (15%), when two 32K-context RULER samples are scheduled simultaneously for prefill, the engine crashes with a DRAM OOM error. All subsequent requests return HTTP 500.

### Environment

| Parameter | Value |
|-----------|-------|
| Model | `google/gemma-3-27b-it` |
| Device | T3K (1x8 mesh) |
| tt_metal_commit | `669abf5` |
| vllm_commit | `22be241` |
| max_model_len | 131072 |
| max_num_seqs | 32 |
| chunked_prefill | disabled (TT devices) |

### Reproduction
```
git checkout adam/gemma-3-tt-fatal-fix
python3 run.py --model gemma-3-27b-it --device t3k --workflow servers --docker-server --dev-mode
python3 run.py --model gemma-3-27b-it --device t3k --workflow evals --limit-samples-mode ci-nightly
```

### Error
After ~27/175 requests, two RULER 32K-context samples are scheduled together for prefill (31,089 + 31,272 = 62,361 tokens). The `all_gather_async` op in `distributed_norm` tries to allocate ~10.5 GB on DRAM that only has ~108 MB free (KV cache at 95.86%):
```
TT_FATAL: Out of Memory: Not enough space to allocate 11274289152 B DRAM buffer across 12 banks (bank size: 1021821792 B, allocated: 908831392 B, free: 112990400 B)
```

### Root Cause
The OOM is caused by commit `104e21cec63`: "Add support for batched prefill same as that of llama 70b for tt-transformers" (PR #35722), introduced between `5aa444a` and `669abf5`.
This commit adds batched prefill logic to `generator.py::prefill_forward_text()`. When 2+ prefill requests share the same padded sequence length, it activates a batched path that allocates tensors for all 32 batch slots (`padded_batch = max_batch_size = max_num_seqs = 32`), even if only 2 users need prefill:
```python
# generator.py line 463-468
use_batched_prefill = (
    batch_size > 1
    and len(set(prefill_seq_lens)) == 1  # same padded length
    and prefill_seq_lens[0] * batch_size < MAX_BATCHED_PREFILL_SEQ_LEN
    and self.data_parallel == 1
)

# line 505: allocates for ALL 32 slots, not just the 2 active users
prefill_ids = torch.zeros(padded_batch, prefill_seq_len, ...)  # [32, 32768]
```
In the model forward pass, this `[32, seq_len]` input is flattened to `[1, 1, 32*seq_len, hidden_dim]`. When `distributed_norm.forward()` calls `all_gather_async`, the output buffer is:
- gemma-3-27b-it: 32 × 32768 × hidden_dim × 2 bytes = 11,274,289,152 B (~10.5 GiB) ← exact match

exceeding available DRAM (92-96% consumed by KV cache).

### Why It Worked Before
At `5aa444a`, the batched prefill code did not exist. Each prefill request was processed sequentially with `batch_size=1`. The `all_gather` buffer for a single 31K-token request was only ~300 MB (27b), easily fitting in DRAM.

### Related
Same OOM pattern observed on gemma-3-4b-it (N300) with the same commits.
```
TT_FATAL: Out of Memory: Not enough space to allocate 5368709120 B DRAM buffer
```
Output buffer: 
- gemma-3-4b-it: 32 × 32768 × 2560 × 2 bytes = 5,368,709,120 B (5.00 GiB) ← also exact match

**Comment by tstescoTT** [comment]
@arobergeTT can you please file a corresponding issue on tt-metal with your summary and repro instructions?

**Comment by arobergeTT** [comment]
tt-metal ticket: https://github.com/tenstorrent/tt-metal/issues/39868

**Comment by bgoelTT** [comment]
Daily:
* @arobergeTT re-requested reviews from codeowners. Failing tt-metal CI is unrelated to his changes and are known on `main`
  * if getting reviews drag on we'll need to reach out directly

**Comment by bgoelTT** [comment]
Closing as PR has been merged https://github.com/tenstorrent/tt-metal/pull/40108
